File: 2002B2/CS/WinForms-Dockman/DockMan/Dockman.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace DockMan
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		private System.Windows.Forms.CheckBox chkLeft;
		private System.Windows.Forms.RadioButton rdbLeft;
		private System.Windows.Forms.CheckBox chkTop;
		private System.Windows.Forms.Panel panel1;
		private System.Windows.Forms.Panel panel2;
		private System.Windows.Forms.RadioButton rdbTop;
		private System.Windows.Forms.CheckBox chkRight;
		private System.Windows.Forms.CheckBox chkBottom;
		private System.Windows.Forms.RadioButton rdbNone;
		private System.Windows.Forms.RadioButton rdbSet;
		private System.Windows.Forms.RadioButton rdbRight;
		private System.Windows.Forms.GroupBox groupBox1;
		private System.Windows.Forms.GroupBox groupBox2;
		private System.Windows.Forms.ToolTip toolTip1;
		private System.Windows.Forms.RadioButton rdbBottom;
		private System.Windows.Forms.Button btnDemo;
		private System.Windows.Forms.RadioButton rdbFill;
		private System.ComponentModel.IContainer components;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
			// Wire up event handlers
			rdbRight.Click += new System.EventHandler(radiobutton_Click);
			rdbFill.Click += new System.EventHandler(radiobutton_Click);
			rdbBottom.Click += new System.EventHandler(radiobutton_Click);
			rdbNone.Click += new System.EventHandler(radiobutton_Click);
			rdbLeft.Click += new System.EventHandler(radiobutton_Click);
			chkTop.Click += new System.EventHandler(checkbox_Click);
			chkLeft.Click += new System.EventHandler(checkbox_Click);
			chkRight.Click += new System.EventHandler(checkbox_Click);
			chkBottom.Click += new System.EventHandler(checkbox_Click);
			rdbTop.Click += new System.EventHandler(radiobutton_Click);

			// Complete intialization of the form
			rdbSet = rdbNone;
			applyChanges();
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		public override void Dispose()
		{
			base.Dispose();
			if(components != null)
				components.Dispose();
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.components = new System.ComponentModel.Container();
			this.chkLeft = new System.Windows.Forms.CheckBox();
			this.rdbLeft = new System.Windows.Forms.RadioButton();
			this.chkTop = new System.Windows.Forms.CheckBox();
			this.panel1 = new System.Windows.Forms.Panel();
			this.panel2 = new System.Windows.Forms.Panel();
			this.rdbTop = new System.Windows.Forms.RadioButton();
			this.chkRight = new System.Windows.Forms.CheckBox();
			this.chkBottom = new System.Windows.Forms.CheckBox();
			this.rdbNone = new System.Windows.Forms.RadioButton();
			this.rdbSet = new System.Windows.Forms.RadioButton();
			this.rdbRight = new System.Windows.Forms.RadioButton();
			this.groupBox1 = new System.Windows.Forms.GroupBox();
			this.groupBox2 = new System.Windows.Forms.GroupBox();
			this.toolTip1 = new System.Windows.Forms.ToolTip(this.components);
			this.rdbBottom = new System.Windows.Forms.RadioButton();
			this.btnDemo = new System.Windows.Forms.Button();
			this.rdbFill = new System.Windows.Forms.RadioButton();
			this.chkLeft.Location = new System.Drawing.Point(8, 48);
			this.chkLeft.Size = new System.Drawing.Size(72, 24);
			this.chkLeft.TabIndex = 2;
			this.chkLeft.Text = "&Left";
			this.rdbLeft.Location = new System.Drawing.Point(8, 72);
			this.rdbLeft.Size = new System.Drawing.Size(72, 24);
			this.rdbLeft.TabIndex = 3;
			this.rdbLeft.Text = "&Left";
			this.chkTop.Location = new System.Drawing.Point(8, 24);
			this.chkTop.Size = new System.Drawing.Size(72, 24);
			this.chkTop.TabIndex = 0;
			this.chkTop.Text = "&Top";
			this.panel1.BackColor = System.Drawing.Color.Green;
			this.panel1.BorderStyle = System.Windows.Forms.BorderStyle.Fixed3D;
			this.panel1.Controls.AddRange(new System.Windows.Forms.Control[] {this.btnDemo});
			this.panel1.Dock = System.Windows.Forms.DockStyle.Fill;
			this.panel1.Size = new System.Drawing.Size(328, 400);
			this.panel1.TabIndex = 1;
			this.panel1.Text = "ButtonPanel";
			this.panel2.BorderStyle = System.Windows.Forms.BorderStyle.Fixed3D;
			this.panel2.Controls.AddRange(new System.Windows.Forms.Control[] {this.groupBox1,
																				 this.groupBox2});
			this.panel2.Dock = System.Windows.Forms.DockStyle.Right;
			this.panel2.Location = new System.Drawing.Point(328, 0);
			this.panel2.Size = new System.Drawing.Size(120, 400);
			this.panel2.TabIndex = 0;
			this.panel2.Text = "ControlsPanel";
			this.toolTip1.SetToolTip(this.panel2, "Resize the form to see the layout effects.");
			this.rdbTop.Location = new System.Drawing.Point(8, 48);
			this.rdbTop.Size = new System.Drawing.Size(72, 24);
			this.rdbTop.TabIndex = 0;
			this.rdbTop.Text = "&Top";
			this.chkRight.Location = new System.Drawing.Point(8, 96);
			this.chkRight.Size = new System.Drawing.Size(72, 24);
			this.chkRight.TabIndex = 1;
			this.chkRight.Text = "&Right";
			this.chkBottom.Location = new System.Drawing.Point(8, 72);
			this.chkBottom.Size = new System.Drawing.Size(72, 24);
			this.chkBottom.TabIndex = 3;
			this.chkBottom.Text = "&Bottom";
			this.rdbNone.Checked = true;
			this.rdbNone.Location = new System.Drawing.Point(8, 24);
			this.rdbNone.Size = new System.Drawing.Size(72, 24);
			this.rdbNone.TabIndex = 5;
			this.rdbNone.TabStop = true;
			this.rdbNone.Text = "&None";
			this.rdbSet.Size = new System.Drawing.Size(100, 23);
			this.rdbSet.TabIndex = 0;
			this.rdbSet.Text = "rdbSet";
			this.rdbRight.Location = new System.Drawing.Point(8, 120);
			this.rdbRight.Size = new System.Drawing.Size(72, 24);
			this.rdbRight.TabIndex = 4;
			this.rdbRight.Text = "&Right";
			this.groupBox1.Controls.AddRange(new System.Windows.Forms.Control[] {this.chkRight,
																					this.chkBottom,
																					this.chkLeft,
																					this.chkTop});
			this.groupBox1.Location = new System.Drawing.Point(16, 16);
			this.groupBox1.Size = new System.Drawing.Size(88, 128);
			this.groupBox1.TabIndex = 0;
			this.groupBox1.TabStop = false;
			this.groupBox1.Text = "A&nchor";
			this.groupBox2.Controls.AddRange(new System.Windows.Forms.Control[] {this.rdbBottom,
																					this.rdbLeft,
																					this.rdbNone,
																					this.rdbRight,
																					this.rdbFill,
																					this.rdbTop});
			this.groupBox2.Location = new System.Drawing.Point(16, 152);
			this.groupBox2.Size = new System.Drawing.Size(88, 176);
			this.groupBox2.TabIndex = 1;
			this.groupBox2.TabStop = false;
			this.groupBox2.Text = "&Dock";
			this.rdbBottom.Location = new System.Drawing.Point(8, 96);
			this.rdbBottom.Size = new System.Drawing.Size(72, 24);
			this.rdbBottom.TabIndex = 1;
			this.rdbBottom.Text = "&Bottom";
			this.btnDemo.Anchor = System.Windows.Forms.AnchorStyles.None;
			this.btnDemo.BackColor = System.Drawing.SystemColors.Control;
			this.btnDemo.FlatStyle = System.Windows.Forms.FlatStyle.Popup;
			this.btnDemo.Location = new System.Drawing.Point(62, 148);
			this.btnDemo.Size = new System.Drawing.Size(120, 24);
			this.btnDemo.TabIndex = 0;
			this.btnDemo.Text = "Demo Button";
			this.toolTip1.SetToolTip(this.btnDemo, "Nothing happens if you click this button.");
			this.rdbFill.Location = new System.Drawing.Point(8, 144);
			this.rdbFill.Size = new System.Drawing.Size(72, 24);
			this.rdbFill.TabIndex = 2;
			this.rdbFill.Text = "&Fill";
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(448, 400);
			this.Controls.AddRange(new System.Windows.Forms.Control[] {this.panel1,
																		  this.panel2});
			this.Location = new System.Drawing.Point(100, 100);
			this.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Show;
			this.Text = "Docking and Anchoring Example";
		}
		#endregion

		private void applyChanges() 
		{

			//Apply Anchoring Settings - maybe multiple
			AnchorStyles nSettings = AnchorStyles.None;
			if (chkTop.Checked)    nSettings |= AnchorStyles.Top;
			if (chkLeft.Checked)   nSettings |= AnchorStyles.Left;
			if (chkBottom.Checked) nSettings |= AnchorStyles.Bottom;
			if (chkRight.Checked)  nSettings |= AnchorStyles.Right;
			btnDemo.Anchor = nSettings ;

			//Apply Docking settings - one only
			if (rdbSet == rdbNone)
				btnDemo.Dock = System.Windows.Forms.DockStyle.None;
			else if (rdbSet == rdbTop)
				btnDemo.Dock = System.Windows.Forms.DockStyle.Top;
			else if (rdbSet == rdbLeft)
				btnDemo.Dock = System.Windows.Forms.DockStyle.Left;
			else if (rdbSet == rdbBottom)
				btnDemo.Dock = System.Windows.Forms.DockStyle.Bottom;
			else if (rdbSet == rdbRight)
				btnDemo.Dock = System.Windows.Forms.DockStyle.Right;
			else // The default is: if (rdbSet is rbFill)
				btnDemo.Dock = System.Windows.Forms.DockStyle.Fill;
		}

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		// <doc>
		// <desc>
		//      Whenever a checkbox is clicked, apply the changes from all
		//      controls.  Note all checkboxes use this same handler.
		// </desc>
		// </doc>
		//
		protected void checkbox_Click(object sender, EventArgs e) 
		{
			applyChanges();
		}

		// <doc>
		// <desc>
		//      Save the radio button that was clicked so we know which one is
		//      checked when we apply the changes.  Note, all radio buttons use
		//      this same click handler.
		// </desc>
		// </doc>
		//
		protected void radiobutton_Click(object sender, EventArgs e) 
		{
			rdbSet = (RadioButton)sender;
			applyChanges();
		}

		private void btnDemo_Click(object sender, System.EventArgs e)
		{

		}
	}
}

File: 2002B2/VB/WinForms-Dockman/DockMan/Dockman.vb
''------------------------------------------------------------------------------
''/ <copyright from='1997' to='2001' company='Microsoft Corporation'>           
''/    Copyright (c) Microsoft Corporation. All Rights Reserved.                
''/       
''/    Source code is intended only as a suppleMent to Microsoft
''/    DevelopMent Tools and/or on-line docuMentation.  See these other
''/    materials for detailed information regarding Microsoft code samples.
''/
''/ </copyright>                                                                
''------------------------------------------------------------------------------
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms


Namespace Microsoft.Samples.Windows.Forms.VB.DockMan

    Public Class DockMan
        Inherits System.Windows.Forms.Form

        'ButtonSet keeps track of which radio button is checked
        Private ButtonSet As System.Windows.Forms.RadioButton

        Public Sub New()

            MyBase.New()

            DockMan = Me

            'This call is required by the Win Form Designer.
            InitializeComponent()

            'TODO: Add any initialization after the InitializeComponent() call

            'Wire up event handlers
            AddHandler rdbRight.Click, New EventHandler(AddressOf Me.radiobutton_Click)
            AddHandler rdbFill.Click, New EventHandler(AddressOf Me.radiobutton_Click)
            AddHandler rdbBottom.Click, New EventHandler(AddressOf Me.radiobutton_Click)
            AddHandler rdbNone.Click, New EventHandler(AddressOf Me.radiobutton_Click)
            AddHandler rdbLeft.Click, New EventHandler(AddressOf Me.radiobutton_Click)
            AddHandler rdbTop.Click, New EventHandler(AddressOf Me.radiobutton_Click)
            AddHandler chkTop.Click, New EventHandler(AddressOf Me.checkbox_Click)
            AddHandler chkLeft.Click, New EventHandler(AddressOf Me.checkbox_Click)
            AddHandler chkRight.Click, New EventHandler(AddressOf Me.checkbox_Click)
            AddHandler Me.Click, New EventHandler(AddressOf Me.checkbox_Click)

            ' Complete intialization of the form
            ButtonSet = rdbNone
            ApplyChanges()

        End Sub

        'Form overrides dispose to clean up the component list.
        Public Overloads Overrides Sub Dispose()
            MyBase.Dispose()
            components.Dispose()
        End Sub

        'The main entry point for the application        
        <STAThread()> Shared Sub Main()
            System.Windows.Forms.Application.Run(New DockMan())
        End Sub

        Private Sub ApplyChanges()

            'Apply Anchoring Settings - maybe multiple
            Dim Settings As AnchorStyles
            Settings = AnchorStyles.None

            If chkTop.Checked Then
                Settings = Settings Or AnchorStyles.Top
            End If

            If chkLeft.Checked Then
                Settings = Settings Or AnchorStyles.Left
            End If

            If chkBottom.Checked Then
                Settings = Settings Or AnchorStyles.Bottom
            End If

            If chkRight.Checked Then
                Settings = Settings Or AnchorStyles.Right
            End If

            btnDemo.Anchor = Settings

            'Apply Docking settings - one only
            If ButtonSet Is rdbNone Then
                btnDemo.Dock = System.Windows.Forms.DockStyle.None
            ElseIf ButtonSet Is rdbTop Then
                btnDemo.Dock = System.Windows.Forms.DockStyle.Top
            ElseIf ButtonSet Is rdbLeft Then
                btnDemo.Dock = System.Windows.Forms.DockStyle.Left
            ElseIf ButtonSet Is rdbBottom Then
                btnDemo.Dock = System.Windows.Forms.DockStyle.Bottom
            ElseIf ButtonSet Is rdbRight Then
                btnDemo.Dock = System.Windows.Forms.DockStyle.Right
            Else ' The default is: if (ButtonSet is rbFill)
                btnDemo.Dock = System.Windows.Forms.DockStyle.Fill
            End If

        End Sub
        Private components As System.ComponentModel.IContainer

#Region " Windows Form Designer generated code "

        'Required by the Windows Form Designer
        Private WithEvents panel1 As System.Windows.Forms.Panel
        Private WithEvents panel2 As System.Windows.Forms.Panel
        Private WithEvents groupBox1 As System.Windows.Forms.GroupBox
        Private WithEvents groupBox2 As System.Windows.Forms.GroupBox
        Private toolTip1 As System.Windows.Forms.ToolTip
        Private WithEvents btnDemo As System.Windows.Forms.Button
        Private WithEvents rdbNone As System.Windows.Forms.RadioButton
        Private WithEvents rdbTop As System.Windows.Forms.RadioButton
        Private WithEvents rdbLeft As System.Windows.Forms.RadioButton
        Private WithEvents rdbBottom As System.Windows.Forms.RadioButton
        Private WithEvents rdbRight As System.Windows.Forms.RadioButton
        Private WithEvents rdbFill As System.Windows.Forms.RadioButton
        Private WithEvents chkTop As System.Windows.Forms.CheckBox
        Private WithEvents chkLeft As System.Windows.Forms.CheckBox
        Private WithEvents chkBottom As System.Windows.Forms.CheckBox
        Private WithEvents chkRight As System.Windows.Forms.CheckBox

        Private WithEvents DockMan As System.Windows.Forms.Form

        'NOTE: The following procedure is required by the Windows Form Designer        
        Private Sub InitializeComponent()
            Me.components = New System.ComponentModel.Container()
            Me.chkLeft = New System.Windows.Forms.CheckBox()
            Me.rdbLeft = New System.Windows.Forms.RadioButton()
            Me.chkTop = New System.Windows.Forms.CheckBox()
            Me.panel1 = New System.Windows.Forms.Panel()
            Me.panel2 = New System.Windows.Forms.Panel()
            Me.rdbTop = New System.Windows.Forms.RadioButton()
            Me.chkRight = New System.Windows.Forms.CheckBox()
            Me.chkBottom = New System.Windows.Forms.CheckBox()
            Me.rdbNone = New System.Windows.Forms.RadioButton()
            Me.ButtonSet = New System.Windows.Forms.RadioButton()
            Me.rdbRight = New System.Windows.Forms.RadioButton()
            Me.groupBox1 = New System.Windows.Forms.GroupBox()
            Me.groupBox2 = New System.Windows.Forms.GroupBox()
            Me.toolTip1 = New System.Windows.Forms.ToolTip(Me.components)
            Me.rdbBottom = New System.Windows.Forms.RadioButton()
            Me.btnDemo = New System.Windows.Forms.Button()
            Me.rdbFill = New System.Windows.Forms.RadioButton()
            Me.chkLeft.Location = New System.Drawing.Point(8, 48)
            Me.chkLeft.Size = New System.Drawing.Size(72, 24)
            Me.chkLeft.TabIndex = 2
            Me.chkLeft.Text = "&Left"
            Me.rdbLeft.Location = New System.Drawing.Point(8, 72)
            Me.rdbLeft.Size = New System.Drawing.Size(72, 24)
            Me.rdbLeft.TabIndex = 3
            Me.rdbLeft.Text = "&Left"
            Me.chkTop.Location = New System.Drawing.Point(8, 24)
            Me.chkTop.Size = New System.Drawing.Size(72, 24)
            Me.chkTop.TabIndex = 0
            Me.chkTop.Text = "&Top"
            Me.panel1.BackColor = System.Drawing.Color.Green
            Me.panel1.BorderStyle = System.Windows.Forms.BorderStyle.Fixed3D
            Me.panel1.Controls.AddRange(New System.Windows.Forms.Control() {Me.btnDemo})
            Me.panel1.Dock = System.Windows.Forms.DockStyle.Fill
            Me.panel1.Size = New System.Drawing.Size(448, 400)
            Me.panel1.TabIndex = 1
            Me.panel1.Text = "ButtonPanel"
            Me.panel2.BorderStyle = System.Windows.Forms.BorderStyle.Fixed3D
            Me.panel2.Controls.AddRange(New System.Windows.Forms.Control() {Me.groupBox1, Me.groupBox2})
            Me.panel2.Dock = System.Windows.Forms.DockStyle.Right
            Me.panel2.Location = New System.Drawing.Point(328, 0)
            Me.panel2.Size = New System.Drawing.Size(120, 400)
            Me.panel2.TabIndex = 0
            Me.panel2.Text = "ControlsPanel"
            Me.toolTip1.SetToolTip(Me.panel2, "Resize the form to see the layout effects.")
            Me.rdbTop.Location = New System.Drawing.Point(8, 48)
            Me.rdbTop.Size = New System.Drawing.Size(72, 24)
            Me.rdbTop.TabIndex = 0
            Me.rdbTop.Text = "&Top"
            Me.chkRight.Location = New System.Drawing.Point(8, 96)
            Me.chkRight.Size = New System.Drawing.Size(72, 24)
            Me.chkRight.TabIndex = 1
            Me.chkRight.Text = "&Right"
            Me.chkBottom.Location = New System.Drawing.Point(8, 72)
            Me.chkBottom.Size = New System.Drawing.Size(72, 24)
            Me.chkBottom.TabIndex = 3
            Me.chkBottom.Text = "&Bottom"
            Me.rdbNone.Checked = True
            Me.rdbNone.Location = New System.Drawing.Point(8, 24)
            Me.rdbNone.Size = New System.Drawing.Size(72, 24)
            Me.rdbNone.TabIndex = 5
            Me.rdbNone.TabStop = True
            Me.rdbNone.Text = "&None"
            Me.ButtonSet.Size = New System.Drawing.Size(100, 23)
            Me.ButtonSet.TabIndex = 0
            Me.ButtonSet.Text = "rdbSet"
            Me.rdbRight.Location = New System.Drawing.Point(8, 120)
            Me.rdbRight.Size = New System.Drawing.Size(72, 24)
            Me.rdbRight.TabIndex = 4
            Me.rdbRight.Text = "&Right"
            Me.groupBox1.Controls.AddRange(New System.Windows.Forms.Control() {Me.chkRight, Me.chkBottom, Me.chkLeft, Me.chkTop})
            Me.groupBox1.Location = New System.Drawing.Point(16, 16)
            Me.groupBox1.Size = New System.Drawing.Size(88, 128)
            Me.groupBox1.TabIndex = 0
            Me.groupBox1.TabStop = False
            Me.groupBox1.Text = "A&nchor"
            Me.groupBox2.Controls.AddRange(New System.Windows.Forms.Control() {Me.rdbBottom, Me.rdbLeft, Me.rdbNone, Me.rdbRight, Me.rdbFill, Me.rdbTop})
            Me.groupBox2.Location = New System.Drawing.Point(16, 152)
            Me.groupBox2.Size = New System.Drawing.Size(88, 176)
            Me.groupBox2.TabIndex = 1
            Me.groupBox2.TabStop = False
            Me.groupBox2.Text = "&Dock"
            Me.rdbBottom.Location = New System.Drawing.Point(8, 96)
            Me.rdbBottom.Size = New System.Drawing.Size(72, 24)
            Me.rdbBottom.TabIndex = 1
            Me.rdbBottom.Text = "&Bottom"
            Me.btnDemo.Anchor = System.Windows.Forms.AnchorStyles.None
            Me.btnDemo.BackColor = System.Drawing.SystemColors.Control
            Me.btnDemo.FlatStyle = System.Windows.Forms.FlatStyle.Popup
            Me.btnDemo.Location = New System.Drawing.Point(64, 72)
            Me.btnDemo.Size = New System.Drawing.Size(120, 24)
            Me.btnDemo.TabIndex = 0
            Me.btnDemo.Text = "Demo Button"
            Me.toolTip1.SetToolTip(Me.btnDemo, "Nothing happens if you click Me button.")
            Me.rdbFill.Location = New System.Drawing.Point(8, 144)
            Me.rdbFill.Size = New System.Drawing.Size(72, 24)
            Me.rdbFill.TabIndex = 2
            Me.rdbFill.Text = "&Fill"
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.ClientSize = New System.Drawing.Size(448, 400)
            Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.panel2, Me.panel1})
            Me.Location = New System.Drawing.Point(100, 100)
            Me.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Show
            Me.Text = "Docking and Anchoring Example"

        End Sub

#End Region

        'Whenever a checkbox is clicked, apply the changes from all
        'controls.  Note all checkboxes use Me saMe handler.
        Private Sub checkbox_Click(ByVal sender As Object, ByVal e As EventArgs)
            ApplyChanges()
        End Sub

        'Save the radio button that was clicked so we know which one is
        'checked when we apply the changes.  Note, all radio buttons use
        'Me saMe click handler.
        Private Sub radiobutton_Click(ByVal sender As Object, ByVal e As EventArgs)
            ButtonSet = CType(sender, RadioButton)
            ApplyChanges()
        End Sub

    End Class

End Namespace





